### ISSUE TITLE
Cannot recover from suspend after recent dom0 update

### ISSUE THREAD

**Issue by esote** [QubesOS/qubes-issues#3901]
### Qubes OS version:

R4

### Affected component(s): 

Resume after suspend

---

### Steps to reproduce the behavior:

1. Updated dom0. I simply ran `qubes-dom0-update`, I am using the default repositories for R4 (current). The update log is as follows:

```
Packages Altered:
    Install  pycairo-1.10.0-5.fc25.x86_64                      @qubes-dom0-cached
    Install  python-xpyb-1.3.1-6.fc24.x86_64                   @qubes-dom0-cached
    Install  python2-nose-1.3.7-11.fc25.noarch                 @qubes-dom0-cached
    Install  python2-pillow-3.4.2-1.fc25.x86_64                @qubes-dom0-cached
    Upgraded python2-qubesadmin-4.0.16-0.1.fc25.noarch         @qubes-dom0-cached
    Upgrade                     4.0.17-0.1.fc25.noarch         @qubes-dom0-cached
    Install  python2-qubesimgconverter-4.0.19-1.fc25.x86_64    @qubes-dom0-cached
    Upgraded python3-qubesadmin-4.0.16-0.1.fc25.noarch         @qubes-dom0-cached
    Upgrade                     4.0.17-0.1.fc25.noarch         @qubes-dom0-cached
    Upgraded python3-qubesdb-4.0.5-1.fc25.x86_64               @anaconda/rawhide
    Upgrade                  4.0.6-1.fc25.x86_64               @qubes-dom0-cached
    Upgraded python3-qubesimgconverter-4.0.17-1.fc25.x86_64    @anaconda/rawhide
    Upgrade                            4.0.19-1.fc25.x86_64    @qubes-dom0-cached
    Upgraded qubes-core-admin-client-4.0.16-0.1.fc25.noarch    @qubes-dom0-cached
    Upgrade                          4.0.17-0.1.fc25.noarch    @qubes-dom0-cached
    Upgraded qubes-core-dom0-4.0.24-1.fc25.x86_64              @qubes-dom0-cached
    Upgrade                  4.0.27-1.fc25.x86_64              @qubes-dom0-cached
    Upgraded qubes-db-4.0.5-1.fc25.x86_64                      @anaconda/rawhide
    Upgrade           4.0.6-1.fc25.x86_64                      @qubes-dom0-cached
    Upgraded qubes-db-dom0-4.0.5-1.fc25.x86_64                 @anaconda/rawhide
    Upgrade                4.0.6-1.fc25.x86_64                 @qubes-dom0-cached
    Upgraded qubes-db-libs-4.0.5-1.fc25.x86_64                 @anaconda/rawhide
    Upgrade                4.0.6-1.fc25.x86_64                 @qubes-dom0-cached
    Upgraded qubes-dbus-1.0.4-1.fc25.noarch                    @anaconda/rawhide
    Upgrade             1.0.5-1.fc25.noarch                    @qubes-dom0-cached
    Upgraded qubes-desktop-linux-common-4.0.12-1.fc25.noarch   @anaconda/rawhide
    Upgrade                             4.0.13-1.fc25.noarch   @qubes-dom0-cached
    Upgraded qubes-desktop-linux-manager-4.0.7-1.fc25.noarch   @anaconda/rawhide
    Upgrade                              4.0.9-1.fc25.noarch   @qubes-dom0-cached
    Upgraded qubes-img-converter-dom0-1.2.4-1.fc25.x86_64      @anaconda/rawhide
    Upgrade                           1.2.5-1.fc25.x86_64      @qubes-dom0-cached
    Upgraded qubes-input-proxy-1.0.10-1.fc25.x86_64            @anaconda/rawhide
    Upgrade                    1.0.11-1.fc25.x86_64            @qubes-dom0-cached
    Upgraded qubes-libvchan-xen-4.0.1-1.fc25.x86_64            @anaconda/rawhide
    Upgrade                     4.0.2-1.fc25.x86_64            @qubes-dom0-cached
    Upgraded qubes-menus-4.0.12-1.fc25.noarch                  @anaconda/rawhide
    Upgrade              4.0.13-1.fc25.noarch                  @qubes-dom0-cached
    Upgraded qubes-mgmt-salt-4.0.8-1.fc25.noarch               @qubes-dom0-cached
    Upgrade                  4.0.9-1.fc25.noarch               @qubes-dom0-cached
    Upgraded qubes-mgmt-salt-admin-tools-4.0.8-1.fc25.noarch   @qubes-dom0-cached
    Upgrade                              4.0.9-1.fc25.noarch   @qubes-dom0-cached
    Upgraded qubes-mgmt-salt-config-4.0.8-1.fc25.noarch        @qubes-dom0-cached
    Upgrade                         4.0.9-1.fc25.noarch        @qubes-dom0-cached
    Upgraded qubes-mgmt-salt-dom0-4.0.8-1.fc25.noarch          @qubes-dom0-cached
    Upgrade                       4.0.9-1.fc25.noarch          @qubes-dom0-cached
    Upgraded qubes-pdf-converter-dom0-2.1.3-1.fc25.x86_64      @anaconda/rawhide
    Upgrade                           2.1.4-1.fc25.x86_64      @qubes-dom0-cached
    Upgraded qubes-release-4.0-1.noarch                        @anaconda/rawhide
    Upgrade                4.0-2.noarch                        @qubes-dom0-cached
    Upgraded qubes-release-notes-4.0-1.noarch                  @anaconda/rawhide
    Upgrade                      4.0-2.noarch                  @qubes-dom0-cached
    Upgraded qubes-usb-proxy-dom0-1.0.17-1.fc25.noarch         @qubes-dom0-cached
    Upgrade                       1.0.18-1.fc25.noarch         @qubes-dom0-cached
    Upgraded qubes-utils-4.0.17-1.fc25.x86_64                  @anaconda/rawhide
    Upgrade              4.0.19-1.fc25.x86_64                  @qubes-dom0-cached
    Upgraded qubes-utils-libs-4.0.17-1.fc25.x86_64             @anaconda/rawhide
    Upgrade                   4.0.19-1.fc25.x86_64             @qubes-dom0-cached
    Install  python2-numpy-1:1.11.2-1.fc25.x86_64              @qubes-dom0-cached
    Install  kernel-1000:4.14.35-1.pvops.qubes.x86_64          @qubes-dom0-cached
    Install  kernel-qubes-vm-1000:4.14.35-1.pvops.qubes.x86_64 @qubes-dom0-cached
```

2. Suspend computer (click suspend from "Log out" prompt, or close laptop lid depending on power saving settings).

### Expected behavior:

Computer shows `xscreensaver`, prompting for password to log back in (per my settings).

### Actual behavior:

Screen just stays black (screen is completely powered off). I am unable to use <kbd>Ctrl+Alt+F2</kbd>. Most times I am able to hold the power button to shut down the machine, but sometimes it does not respond to holding the power button (seems to keep going back and forth between states?, to solve this I just wait a bit and try again until it works).

### General notes:

I am using the fallback UEFI bootloader in `/boot/efi/EFI/BOOT` -- which requires changing upon updating the Linux Kernel or Xen. Therefore, after updating the kernel, I copied the contents of `/boot/efi/EFI/qubes/*` to `/boot/efi/EFI/BOOT/*` and renamed `xen.cfg` to `BOOTX64.cfg` and `xen.efi` to `BOOTX64.efi`.

```
[root@dom0 EFI]# pwd
/boot/efi/EFI
[root@dom0 EFI]# tree .
.
├── BOOT
│   ├── BOOTX64.cfg
│   ├── BOOTX64.efi
│   ├── initramfs-4.14.18-1.pvops.qubes.x86_64.img
│   ├── initramfs-4.14.35-1.pvops.qubes.x86_64.img
│   ├── vmlinuz-4.14.18-1.pvops.qubes.x86_64
│   ├── vmlinuz-4.14.35-1.pvops.qubes.x86_64
│   ├── xen-4.8.3.efi
│   └── xen.cfg.anacbak
└── qubes
    ├── initramfs-4.14.18-1.pvops.qubes.x86_64.img
    ├── initramfs-4.14.35-1.pvops.qubes.x86_64.img
    ├── vmlinuz-4.14.18-1.pvops.qubes.x86_64
    ├── vmlinuz-4.14.35-1.pvops.qubes.x86_64
    ├── xen-4.8.3.efi
    ├── xen.cfg
    ├── xen.cfg.anacbak
    └── xen.efi

2 directories, 16 files
```

This is the contents of `/boot/efi/EFI/BOOT/BOOTX64.cfg` (leaving out `rd.luks.uuid`):

```
[global]
default=4.14.35-1.pvops.qubes.x86_64

[4.14.18-1.pvops.qubes.x86_64]
options=loglvl=all dom0_mem=min:1024M dom0_mem=max:4096M iommu=no-igfx ucode=scan efi=no-rs
kernel=vmlinuz-4.14.18-1.pvops.qubes.x86_64 root=/dev/mapper/qubes_dom0-root rd.lvm.lv=qubes_dom0/root rd.lvm.lv=qubes_dom0/swap i915.alpha_support=1 rhgb quiet rd.qubes.hide_all_usb
ramdisk=initramfs-4.14.18-1.pvops.qubes.x86_64.img
[4.14.35-1.pvops.qubes.x86_64]
options=loglvl=all dom0_mem=min:1024M dom0_mem=max:4096M iommu=no-igfx ucode=scan efi=no-rs
kernel=vmlinuz-4.14.35-1.pvops.qubes.x86_64 root=/dev/mapper/qubes_dom0-root rd.lvm.lv=qubes_dom0/root rd.lvm.lv=qubes_dom0/swap i915.alpha_support=1 rhgb quiet rd.qubes.hide_all_usb
ramdisk=initramfs-4.14.35-1.pvops.qubes.x86_64.img
```

---

### Related issues:

#3705 -- different because I was **not** experiencing this issue before I updated dom0. I still have Xen 4.8.3-3, and Xen was not upgraded in the update.

**Comment by marmarek** [comment]
Does it work if you change back to 4.14.18 kernel? You can do that in xen.cfg/BOOTX64.cfg.

**Comment by Wikinaut** [comment]
To the Qubes Team: in addition to the report of @esote , today a dom0 upgrade crashed (destroyed apparently) my system, so that I cannot boot any more (bootloader?), the NUC i5 does not find a bootable device anymore.

**Comment by esote** [comment]
@marmarek No, I tried switching back to 4.14.18 (verified via `uname -r` after reboot), and still had the issue.

**Comment by Wikinaut** [comment]
@marmarek I have not checked that; I updated about three times in the past weeks. Today, I saw many errors in the update terminal regarding keys and so on, then it stopped in the middle. I aborted that (pressed many times ctrl-c), rebooted and: dead. May be, that the boot drive disk space was exhausted....

.....but this should be checked! Perhaps an issue should be filed - I cannot decide, because I have had a lot of experience with the 3.2, but the 4.0 is much more difficult to use. (in my view)

**Comment by esote** [comment]
@marmarek The system is definitely suspending. I know this because I have a light on the laptop that blinks orange when it is suspended.

I tried switching the default kernel to 4.14.18 (verified sys-net and sys-usb). I still have the issue (even if all VMs are powered off, etc.), and after restarting the machine to double-check.

@Wikinaut Maybe open a separate issue for this, it seems to be different than the one I am receiving.

**Comment by Wikinaut** [comment]
@esote Yes (some parts looked similar, so my system was also "locked", I had to remove power and so on).

@marmarek I will report about a problem with TemplateVMs in a different issue.

**Comment by evilaliv3** [comment]
I report this same issue as described in the same comment on a Thinkpad T480 and a fresh installed Qubes 4

After suspend the system resume slowly, the xscreen screensaver remain frozen with impossibility to type or change screen.

I managed to get the suspend to work by removing the USB3 controller from sys-usb;
Obviously removing the USB3 controller i'm loosing possibility to attach USB devices so that this represents just a short term fix to enable suspend/resume to work correctly but proper fix should still be identified.

**Comment by evilaliv3** [comment]
As an update i managed to get suspend to work with the following setup:
- usb xhci controller reattached (so that usb works)
- rmmod xhci_pci
- rmmod ehci_pci

This way all resume correctly, then the machine froze only when i reload the module xhci_pci

**Comment by evilaliv3** [comment]
An alternative working completely is:
- shut down sys-usb
- sleep
- awake
- startup sys-usb

this procedure works with no problems.

**Comment by evilaliv3** [comment]
@marmarek: is there a way to automate the above procedure?

**Comment by esote** [comment]
@marmarek I was able to solve this issue (cannot recover from suspend & system failure [machine reboots] on suspend) by starting sys-usb before suspending. It therefore appears to be a duplicate of #3689, so I will close this issue.

**Comment by andrewdavidwong** [comment]
Duplicate of #3689

**Comment by evilaliv3** [comment]
@marmarek : I just found out that on the t480 , after a failure to resume with blank screen, if one connect and disconnect the power plug the laptop does correctly resume.

Have you ever experienced this? Maybe this could be an hint on what could be causing the failure and how the issue could be fixed?

**Comment by andrewdavidwong** [comment]
> @marmarek : I just found out that on the t480 , after a failure to resume with blank screen, if one connect and disconnect the power plug the laptop does correctly resume.
> 
> Have you ever experienced this? Maybe this could be an hint on what could be causing the failure and how the issue could be fixed?

This issue is closed as a duplicate of #3689, so you might want to comment there instead (unless you can show that this isn't actually a duplicate of #3689).
